engines/advancedDetector.cpp

```diff
@@ -575,7 +575,11 @@ const char *md5PropToGameFile(MD5Properties flags) {
 static bool getFilePropertiesIntern(uint md5Bytes, const AdvancedMetaEngine::FileMap &allFiles, MD5Properties md5prop, const Common::String &fname, FileProperties &fileProps);
 
 bool AdvancedMetaEngineDetection::getFileProperties(const FileMap &allFiles, MD5Properties md5prop, const Common::String &fname, FileProperties &fileProps) const {
-	Common::String hashname = Common::String::format("%s:%s:%d", md5PropToCachePrefix(md5prop), fname.c_str(), _md5Bytes);
+	Common::String hashname = md5PropToCachePrefix(md5prop);
+		hashname += ':';
+		hashname += fname;
+		hashname += ':';
+		hashname += Common::String::format("%d", _md5Bytes);
```

Comment:
Could you try using `itoa()` here and see if it is any faster?

Reply:
`itoa` is a non-standard function. However out of curiosity I have tried `std::to_string` and my custom `itoa`:

- `Common::String::format`: 74.290s 
- `std::to_string`: 73.440s
- my own `itoa`: 73.070s

```C
static Common::String itoa(uint val)
{
	Common::String s;

	do
	{
		uint a = val % 10;
		val /= 10;

		s += a + '0';

	} while (val > 0);

	if (s.size() > 1) {
		int start = 0 , end = s.size()-1;
		while (start < end) {
			auto temp = s[start];
			s.setChar(s[end], start);
			s.setChar(temp, end);
			start += 1;
			end -= 1;
		}
	}

	return s;
}
```

So there is some speedup but not _that_ significant.

Reply:
Currently it's not an itoa but more an utoa.
I think this could go into BaseString class.

Reply:
Yes, that would be nice to have in the base string class. Just to my surprise, the code formatting is off.

Reply:
So if I implemented BaseString::itoa() (which would handle also negative numbers), would you accept it?